python_modules/dagster/dagster/_core/definitions/asset_daemon_context.py

```diff
@@ -353,9 +352,12 @@ def evaluate_asset(
                 to_materialize.update(asset_partitions)
             self._logger.debug("Done evaluating materialize rule")
 
-        # These should be conditions, but aren't currently, so we just manually strip out things
+        # these should be conditions, but aren't currently, so we just manually strip out things
         # from our materialization set
         for candidate in list(to_materialize):
+            # ignore any partitions that were skipped on the last tick
```

Comment:
What if a candidate that was skipped on the last tick would no longer be skipped on this tick? E.g. if we have a skip rule that avoids materializing anything before 9 am, and the last tick is before 9 am and the next tick is after 9 am?

Reply:
that comment is confusing and I'll update it, but that line is just saying "do not check things that we skipped last tick for the weird 'not really discard rules' of 'currently being backfilled' and 'has invalid parents', because if they were valid last tick they're still valid now." basically just a performance boost (as it can be a bit expensive to check that all of the parents of a bunch of skipped asset partitions are still valid).

Reply:
Still not 100% grokking - what if a backfill starts in between the last tick and the current tick? We'd want to remove it from `to_materialize` in that case, right?

Reply:
Just circling back on this one, what's going on here is just a jank version of the candidate filtering that happens in other "actual" rules. You're correct that for the backfill case, we'd want to have different candidate filtering behavior vs. the missing parent partition case.

I updated the code to reflect that, and once Claire's changes go in, we can move that candidate filtering stuff to the rules.